File: sdks/javascript/src/anthropic.ts
// Anthropic SDK adapter for @wmcp/sdk.
import type { Tool } from "./index";

export interface AnthropicTool {
  name: string;
  description: string;
  input_schema: { type: "object"; properties: Record<string, any>; required?: string[] };
}

export function toAnthropicTools(tools: Tool[]): AnthropicTool[] {
  return tools.map((t) => ({
    name: t.name,
    description: t.description,
    input_schema:
      (t.inputSchema as AnthropicTool["input_schema"]) || {
        type: "object",
        properties: {},
      },
  }));
}

/**
 * Convenience runner — pass a `tool_use` block from Claude's response, get
 * wmcp.sh to execute it. Pass a WmcpClient instance and the URL.
 *
 * @example
 *   const msg = await anthropic.messages.create({ tools: toAnthropicTools(tools), … });
 *   for (const block of msg.content) {
 *     if (block.type === "tool_use") {
 *       const result = await executeToolUse(client, url, block);
 *     }
 *   }
 */
export async function executeToolUse(
  client: { execute: (url: string, tool: string, args: any) => Promise<any> },
  url: string,
  toolUse: { name: string; input?: any }
) {
  return client.execute(url, toolUse.name, toolUse.input || {});
}

File: sdks/python/wmcp/anthropic.py
"""Anthropic SDK adapter — turn wmcp Tools into the shape `anthropic.messages.create(tools=…)` wants."""

from __future__ import annotations

from typing import Any, Dict, List

from .client import Tool


def to_anthropic_tools(tools: List[Tool]) -> List[Dict[str, Any]]:
    """Convert wmcp tools to Anthropic's tool_use schema.

    Usage:

        from anthropic import Anthropic
        from wmcp import WmcpClient
        from wmcp.anthropic import to_anthropic_tools

        client = WmcpClient()
        wmcp_tools = client.tools("https://www.allbirds.com/products/mens-wool-runners")

        anthropic = Anthropic()
        msg = anthropic.messages.create(
            model="claude-opus-4-7",
            max_tokens=1024,
            tools=to_anthropic_tools(wmcp_tools),
            messages=[{"role": "user", "content": "Add size 10 to my cart."}],
        )
    """
    out: List[Dict[str, Any]] = []
    for t in tools:
        out.append(
            {
                "name": t.name,
                "description": t.description,
                "input_schema": t.input_schema,
            }
        )
    return out


def execute_tool_use(client, url: str, tool_use: Dict[str, Any]) -> Dict[str, Any]:
    """Convenience: take an Anthropic `tool_use` content block, run it via wmcp."""
    return client.execute(url=url, tool=tool_use["name"], args=tool_use.get("input", {}))
